scipy/optimize/_linprog_ip.py

```diff
@@ -0,0 +1,1940 @@
+"""
+An interior-point method for linear programming.
+"""
+# Author: Matt Haberland
+
+from __future__ import print_function, division, absolute_import
+import numpy as np
+import scipy as sp
+import scipy.sparse as sps
+from warnings import warn
+from scipy.linalg import LinAlgError
+from .optimize import OptimizeResult, OptimizeWarning, _check_unknown_options
+#from scipy.optimize import _check_unknown_options
+from scipy.optimize._remove_redundancy import _remove_redundancy
+
+def _clean_inputs(
+        c,
+        A_ub=None,
+        b_ub=None,
+        A_eq=None,
+        b_eq=None,
+        bounds=None):
+    """
+    Given user inputs for a linear programming problem, return the
+    objective vector, upper bound constraints, equality constraints,
+    and simple bounds in a preferred format.
+
+    Parameters
+    ----------
+    c : array_like
+        Coefficients of the linear objective function to be minimized.
+    A_ub : array_like, optional
+        2-D array which, when matrix-multiplied by x, gives the values of the
+        upper-bound inequality constraints at x.
+    b_ub : array_like, optional
+        1-D array of values representing the upper-bound of each inequality
+        constraint (row) in A_ub.
```

Comment:
I tried to follow the existing `linprog` as an example, but I don't think it was perfectly consistent. I assume you don't just mean around `A_ub` here but around all the other variable names, too? Everywhere or just the public functions/classes? 

On the left of the `:` as well as on the right? e.g.
```
    b_ub : array_like, optional
        1-D array of values representing the upper-bound of each inequality
        constraint (row) in ``A_ub``.
```
or
```
    ``b_ub`` : array_like, optional
        1-D array of values representing the upper-bound of each inequality
        constraint (row) in ``A_ub``.
```